client/reader/recent/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { Button } from '@wordpress/components';
+import { DataViews, SupportedLayouts, View } from '@wordpress/dataviews';
+import { translate } from 'i18n-calypso';
+import { useState, useEffect, useCallback } from 'react';
+import { useSelector, shallowEqual, useDispatch } from 'react-redux';
+import { AnyAction } from 'redux';
+import { ThunkDispatch } from 'redux-thunk';
+import { getPostByKey } from 'calypso/state/reader/posts/selectors';
+import { requestPage } from 'calypso/state/reader/streams/actions';
+import { viewStream } from 'calypso/state/reader-ui/actions';
+import type { AppState } from 'calypso/types';
+import './style.scss';
+
+interface ReaderPost {
+	site_name: string;
+	postId: number;
+	feedId: number;
+}
+
+const Recent = () => {
+	const dispatch = useDispatch< ThunkDispatch< AppState, void, AnyAction > >();
+
+	const [ selectedItem, setSelectedItem ] = useState< ReaderPost | null >( null );
+
+	const [ view, setView ] = useState( {
+		type: 'list',
+		fields: [ 'post' ],
+	} );
+
+	const { data, post } = useSelector(
+		( state: AppState ) => ( {
+			data: state.reader?.streams?.following,
+			post: selectedItem
+				? getPostByKey( state, {
+						feedId: +selectedItem.feedId,
+						postId: +selectedItem.postId,
+				  } )
+				: null,
+		} ),
+		shallowEqual
+	);
+
+	const fields = [
+		{
+			id: 'post',
+			label: translate( 'Blog' ),
+			render: ( { item }: { item: ReaderPost } ) => {
+				return <Button onClick={ () => setSelectedItem( item ) }>{ item.site_name }</Button>;
+			},
+			enableHiding: false,
+		},
+	];
+
+	const defaultLayouts = [
+		{
+			label: translate( 'List' ),
+			icon: 'list-view',
+		},
+	];
+
+	const fetchData = useCallback( () => {
+		dispatch( viewStream( 'following', window.location.pathname ) as AnyAction );
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
We had to make an exception for this in Reader Onboarding too. Maybe we need to look at the typing on state management.